src/rootserver/balance/ob_balance_group_info.h

```diff
@@ -55,60 +56,139 @@ class ObTransferPartGroup
   share::ObTransferPartList part_list_;
 };
 
+typedef common::ObArray<ObTransferPartGroup *> ObPartGroupBucket;
+
+class ObBalanceGroupUnit
+{
+public:
+  explicit ObBalanceGroupUnit(common::ObIAllocator &alloc):
+    inited_(false),
+    alloc_(alloc),
+    part_group_buckets_(OB_MALLOC_NORMAL_BLOCK_SIZE, ModulePageAllocator(alloc, "PGBuckets")),
+    part_group_cnt_(0)
+  {}
+  ~ObBalanceGroupUnit();
+  int init(const int64_t bucket_num);
+  bool is_valid() const { return inited_; }
+  int64_t get_part_group_count() const { return part_group_cnt_; }
+  int append_part_group(const uint64_t part_group_uid, ObTransferPartGroup *const part_group);
+  int append_part_group(const int64_t bucket_idx, ObTransferPartGroup *const part_group);
```

Comment:
1.Functions that are not exposed to the public need to be set to private.
2.Try not to set the same function name, it is easy to be misused.